src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -63,9 +63,13 @@ bool ClusterTaskManager::SchedulePendingTasks() {
   bool did_schedule = false;
   for (auto shapes_it = tasks_to_schedule_.begin();
        shapes_it != tasks_to_schedule_.end();) {
+    auto sched_cls = shapes_it->first;
     auto &work_queue = shapes_it->second;
+    auto max_running_tasks = MaxRunningTasksPerSchedulingClass(sched_cls);
     bool is_infeasible = false;
-    for (auto work_it = work_queue.begin(); work_it != work_queue.end();) {
+    for (auto work_it = work_queue.begin();
```

Comment:
Can you add a comment here to explain why we cap tasks by scheduling class? It's nice to have the explanation inline so we don't have to go to github to understand the code.